core/domain/question_services.py

```diff
@@ -338,11 +341,11 @@ def get_question_summaries_and_skill_descriptions(
         a time is not supported currently.
 
     Returns:
-        list(QuestionSummary), list(str), str|None. The list of question
-            summaries and the corresponding linked skill descriptions which are
-            linked to the given skill ids and the next cursor value to be used
-            for the next batch of questions (or None if no more pages are left).
-            The returned next cursor value is urlsafe.
+        list(QuestionSummary), list(list(str)), str|None. The list of question
```

Comment:
I think we are coupling two things together too much throughout the stack. Can we fetch the skill descriptions separately to the question summaries? I think that will make a bunch of things simpler and avoid methods like get_A_and_B.

(I came to this suggestion by thinking "hm, we really should have a domain object for this composite of skill description and question summaries", and then realizing that it doesn't really make sense as a concept.)

Reply:
Hmm but how does that make things simpler? We would still need to deduplicate like this right?

Reply:
Did you mean simply separate them into two functions?

Reply:
Yes.

Reply:
Flagging this comment as unresolved

Reply:
Done.